Grbl_Esp32.sln

```diff
@@ -0,0 +1,31 @@
+﻿
```

Comment:
I think it would be better if we make the babysteps even smaller, splitting this patch into the following series:
* Add .clang-format and use it to format the minimally-edited code. By minimally, I mean "the only changes are the addition of // clang-format off (and on) where necessary"
* Add VS support files
* Add coding style file(s)
* Change header guards to #pragma once  (is clang-format okay with header guards or do we have to do the pragma thing at the same time as the formatting?)
* Rename files to be consistent with their included classes
* Get rid of .cpp includes - BUT - I am starting to remember why we did it that way.  It has to do with the &*(#$ Arduino IDE.  It won't look in subdirectories for .cpp files - it treats subdirectories as Arduino Libraries. One solution might be to get rid of the subdirectory structure and move everything up into the top directory.
The individual patches will be easier to review and test.